pkg/actuators/machine/machine.go

```diff
@@ -0,0 +1,321 @@
+	instance, err := ms.ovirtClient.GetVmByMachine(*ms.machine)
+	if err != nil {
+		return err
+	}
+	if instance == nil {
+		klog.Infof("Skipped deleting a VM that is already deleted.\n")
+		return nil
+	}
+
+	return ms.ovirtClient.DeleteVM(instance.MustId())
+}
+
+// returns the ignition from the userData secret
+func (ms *machineScope) getIgnition() ([]byte, error) {
+	if ms.machineProviderSpec == nil || ms.machineProviderSpec.UserDataSecret == nil {
+		return nil, nil
+	}
+	userDataSecret := &corev1.Secret{}
+	objKey := client.ObjectKey{
+		Namespace: ms.machine.Namespace,
+		Name:      ms.machineProviderSpec.UserDataSecret.Name,
+	}
+	if err := ms.client.Get(ms.Context, objKey, userDataSecret); err != nil {
+		return nil, err
+	}
+	userData, exists := userDataSecret.Data[userDataSecretKey]
+	if !exists {
+		return nil, fmt.Errorf("secret %s missing %s key", objKey, userDataSecretKey)
+	}
+	return userData, nil
+}
+
+func (ms *machineScope) patchMachine(ctx context.Context, condition ovirtconfigv1.OvirtMachineProviderCondition) error {
+	instance, err := ms.ovirtClient.GetVmByMachine(*ms.machine)
+	if err != nil {
+		return err
+	}
+	ms.reconcileMachineProviderID(instance)
+	klog.V(5).Infof("Machine %s provider status %s", instance.MustName(), instance.MustStatus())
+
+	err = ms.reconcileMachineNetwork(ctx, instance)
+	if err != nil {
+		return err
+	}
+	ms.reconcileMachineAnnotations(instance)
+	err = ms.reconcileMachineProviderStatus(instance, condition)
+	if err != nil {
+		return err
+	}
+
+	// Copy the status, because its discarded and returned fresh from the DB by the machine resource update.
+	// Save it for the status sub-resource update.
+	statusCopy := *ms.machine.Status.DeepCopy()
+	klog.Info("Updating machine resource")
+
+	// TODO the namespace should be set on actuator creation. Remove the hardcoded openshift-machine-api.
+	newMachine, err := ms.machinesClient.Machines("openshift-machine-api").Update(context.TODO(), ms.machine, metav1.UpdateOptions{})
+	if err != nil {
+		return err
+	}
+
+	newMachine.Status = statusCopy
+	klog.Info("Updating machine status sub-resource")
+	if _, err := ms.machinesClient.Machines("openshift-machine-api").UpdateStatus(context.TODO(), newMachine, metav1.UpdateOptions{}); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (ms *machineScope) reconcileMachineNetwork(ctx context.Context, instance *ovirtC.Instance) error {
+	switch instance.MustStatus() {
+	// expect IP addresses only on those statuses.
+	// in those statuses we 'll try reconciling
+	case ovirtsdk.VMSTATUS_UP, ovirtsdk.VMSTATUS_MIGRATING:
+		break
+
+	// update machine status.
+	case ovirtsdk.VMSTATUS_DOWN:
+		return nil
+
+	// return error if vm is transient state this will force retry reconciling until VM is up.
+	// there is no event generated that will trigger this.  BZ1854787
+	default:
+		return fmt.Errorf("Aborting reconciliation while VM %s  state is %s", instance.MustName(), instance.MustStatus())
```

Comment:
This error message seems to be less than helpful, consider rewording it.

Reply:
Done